### PR TITLE
Add range slider support

### PR DESCRIPTION
**Pull request by embeddedt**
This PR adds support for range sliders, as well as moving the start point of an `lv_bar` to a custom value.

Test code:

```c
	lv_obj_t * slider = lv_slider_create(lv_scr_act(), NULL);
	lv_obj_align(slider, NULL, LV_ALIGN_CENTER, 0, 0);
	lv_slider_set_type(slider, LV_SLIDER_TYPE_RANGE);
	lv_slider_set_value(slider, 75, LV_ANIM_OFF);
	lv_slider_set_left_value(slider, 25, LV_ANIM_OFF);
```

### CODE REVIEW

File: src/lv_objx/lv_bar.h [hunk 1]
```diff
@@ -111,13 +119,27 @@ void lv_bar_set_value(lv_obj_t * bar, int16_t value, lv_anim_enable_t anim);
...
+
 /**
  * Make the bar symmetric to zero. The indicator will grow from zero instead of the minimum
  * position.
  * @param bar pointer to a bar object
  * @param en true: enable disable symmetric behavior; false: disable
+ * @deprecated As of v7.0, you should use `lv_bar_set_type` instead.
  */
-void lv_bar_set_sym(lv_obj_t * bar, bool en);
+static inline void lv_bar_set_sym(lv_obj_t * bar, bool en)
```
Comment (kisvegabor) [inline src/lv_objx/lv_bar.h]: Wouldn't be better have a `lv_bar_set_type(lv_obj_t * bar, lv_bar_type_t type)` function?

File: src/lv_objx/lv_slider.h [hunk 2]
```diff
@@ -173,11 +221,28 @@ static inline uint16_t lv_slider_get_anim_time(lv_obj_t * slider)
...
+		return LV_SLIDER_TYPE_NORMAL;
 }
 
+/**
+ * Get whether the slider is symmetric or not.
+ * @param slider pointer to a slider object
+ * @return true: symmetric is enabled; false: disable
+ * @deprecated As of v7.0, you should use `lv_slider_get_type` instead.
+ */
+static inline bool lv_slider_get_sym(lv_obj_t * slider) {
```
Comment (kisvegabor) [inline src/lv_objx/lv_slider.h]: IMO it's not required if we have `lv_slider_get_type`.

**Review by kisvegabor — COMMENTED** [review]
I've just quickly checked it and it looks good!

I've only a few questions about the API.

File: src/lv_objx/lv_bar.h (hunk 1 above)
Comment (embeddedt) [reply src/lv_objx/lv_bar.h]: It exists:

https://github.com/littlevgl/lvgl/blob/5cd58c6f6f6cdb4e5a9b20b6bdd4acacfc9619c3/src/lv_objx/lv_bar.h#L122-L127

The `set_sym` and `get_sym` functions are there for backwards compatibility (to save people from modifying their code). If you think they're not necessary I can just remove them.

Comment (kisvegabor) [reply src/lv_objx/lv_bar.h]: >  If you think they're not necessary I can just remove them.

I see. 
If I needed to choose between minor API break and duplication in the API, I'd choose the minor API break.

We can still add `lv_bar/slider_set/get_sym` to [lv_api_map.h](https://github.com/littlevgl/lvgl/blob/dev-7.0/src/lv_api_map.h).

Comment (embeddedt) [reply src/lv_objx/lv_bar.h]: I've moved them there.

**Comment by kisvegabor** [comment]
Great! 
Do you think it's ready to merge?

**Comment by embeddedt** [comment]
Yes. I will merge it to `dev-7.0`.

**Comment by flossandmeditate** [comment]
I'll be heavily testing this delightful feature over the next few days, how would you like to receive bugs/observations?

**Comment by embeddedt** [comment]
Just comment here.

**Comment by flossandmeditate** [comment]
Here's a bug for regular slider that might be related to this feature being added...
For the two sliders in the attached image. They are draggable between the correct numbers are set with...
`lv_slider_set_range(LEDBrightnessSlider, 1 , 10);`
`lv_slider_set_range(InstrumentSelectSlider, 1 , 85); `

However as you can see the slider has inserted gap/space for a 0 value on the left of the slider. Can't slide here. It is barely noticeable is range is 1 to 100, but if range is 1 to 10 it is very noticeable....

sliderstartodd

**Comment by embeddedt** [comment]
I probably broke it when implementing range slider support. I'll take a look.

**Comment by embeddedt** [comment]
@flossandmeditate Nice catch! I've fixed it in https://github.com/littlevgl/lvgl/commit/73cb9d45d789bcb12843437e769eac97d40c6c04.

**Pull request merged**